hs_core/models.py

```diff
@@ -1898,6 +1900,35 @@ def delete_all_elements(self):
         self.relations.all().delete()
         self.funding_agencies.all().delete()
 
+    def db_instance_to_dict(self, instance, fields=None, exclude=None):
+        """
+        modified version of Django's model_to_dict method accomodates many_to_many relations
+
+        Returns a dict containing the data in ``instance`` suitable for passing as
+        a Form's ``initial`` keyword argument.
+
+        ``fields`` is an optional list of field names. If provided, only the named
+        fields will be included in the returned dict.
+
+        ``exclude`` is an optional list of field names. If provided, the named
+        fields will be excluded from the returned dict, even if they are listed in
+        the ``fields`` argument.
+        """
+        opts = instance._meta
+        data = {}
+        for f in chain(opts.concrete_fields, opts.many_to_many):
```

Comment:
@jsadler2 This ```for``` loop is different from django native ```model_to_dict()``` function. In django version 1.8 we are using now, the ```for``` loop is ```for f in chain(opts.concrete_fields, opts.virtual_fields, opts.many_to_many):```. In the latest django version, this ```for``` loop is ```for f in chain(opts.concrete_fields, opts.private_fields, opts.many_to_many):```. As you can see, this API function could be changed within Django from version to version, and customizing this Django's native function with our own copied version with a little bit customization leaves us vulnerable for potential breaking code in the future when we upgrade Django version. So I think overriding the default native Django ```model_to_dict``` native function is the last resort we would go with only if we could not find any better solution.  I think the best solution for this issue is to see if our use of ManyToManyField can use the returned primary keys to create new elements for ManyToManyField. I'll add @aphelionz and @pkdash into this thread to get their inputs as well. @aphelionz @pkdash You can see our previous comments on this issue from @jsadler2 and me for background on this issue. It'd be great if you could chime in and provide specific ideas and solutions if possible.